autoload/go/lint.vim

```diff
@@ -95,6 +95,87 @@ function! go#lint#Gometa(bang, autosave, ...) abort
   endif
 endfunction
 
+function! go#lint#Golangci(bang, autosave, ...) abort
+  if a:0 == 0
+    let goargs = [expand('%:p:h')]
+  else
+    let goargs = a:000
+  endif
+
+  if empty(go#config#GolangciLintCommand())
+    let bin_path = go#path#CheckBinPath("golangci-lint")
+    if empty(bin_path)
+      return
+    endif
+
+    let cmd = [bin_path]
+    let cmd += ["run"]
+    let cmd += ["--print-issued-lines=false"]
+    let cmd += ["--fast"]
+
+    " linters
+    let linters = a:autosave ? go#config#GolangciLintAutosaveEnabled() : go#config#GolangciLintEnabled()
+    for linter in linters
+      let cmd += ["--enable=".linter]
+    endfor
+
+    for linter in go#config#GolangciLintDisabled()
+      let cmd += ["--disable=".linter]
```

Comment:
`--disable` conflicts with the `--disable-all` flag when run together, so need to hash out some more things around that